superdesk/errors.py

```diff
@@ -558,5 +558,25 @@ def httpPushError(cls, exception=None, destination=None):
         return PublishHTTPPushError(14000, exception, destination)
 
 
+class PublishHTTPPushClientError(PublishHTTPPushError):
+    _codes = {
+        14001: "HTTP push publish client error",
+    }
+
+    @classmethod
+    def httpPushError(cls, exception=None, destination=None):
+        return PublishHTTPPushClientError(14001, exception, destination)
+
+
+class PublishHTTPPushServerError(PublishHTTPPushError):
+    _codes = {
+        14002: "HTTP push publish server error",
+    }
+
+    @classmethod
```

Comment:
exception handling should be refactored indeed, it started to piss me off first when i were working on some ingest stuff for VPP.

apart from the things @naomiaro mentioned already, it's really annoying what the original context of exception is usually lost, it should print trace to the logs.

and regarding the discussed topic -- why not to make exceptions inherit from some base class instead of doing this classmethod hack?